api/database/migrations/sql/20190808_create_user_session_monitoring_table.sql

```diff
@@ -0,0 +1,26 @@
+/*
+ * Create user session monitoring table
+ */
+
+CREATE TABLE user_session_record (
+  user_session_record_id SERIAL NOT NULL UNIQUE,
+  user_account_id        INT NOT NULL,
+  organisation_id        INT NOT NULL,
+  session_id             VARCHAR NOT NULL UNIQUE,
+  session_end_type       VARCHAR,
+  referrers              JSONB NOT NULL,
```

Comment:
This is a list of clientside routes that the session has accessed? It may be worth renaming this field to clearly describe this rather than being tied to request headers.

Also it only tells us about single access rather than quantity, not sure this data is useful in this format.

Reply:
> Also it only tells us about single access rather than quantity, not sure this data is useful in this format.

It's useful but it can answer fewer questions than storing (for example) the number of times each page was accessed. More complex quantities/statistics were not specified in the user story, but tbh I can see it being potentially better for us to just store all the request headers for each request (stripping out useless stuff) and then worrying about what can be calculated from that later.

If that's the way forward I'll rename this column to `request_headers`, and it'll just be a `Dictionary<{}>[]`.